notes/force-move-implementation.md

```diff
@@ -0,0 +1,178 @@
+# ForceMove Implementation Notes
+
+Active set of notes explaining the ForceMove implementations.
+
+The goal is to make a close-to-optimal implementation of the force-move protocol on the EVM.
+
+## How to get the outcome from a finalized channel
+
+While channel settlement is out of the scope of this piece of work, we will need to interface with the channel funding and settlement layer at a later point.
+
+The strategy here will be to have a method that can be called when a channel is finalized, which will be responsible for interfacing. The precise details of this method will depend on the settlement layer. A few options here include (a) having an `assertOutcome` method which is passed an outcome and returns true if that is the outcome that was finalized or (b) having a `pushOutcome` method, which pushes the outcome into one or more other contracts in the settlement layer.
+
+_Q: Is it inefficient to require a method call for this purpose?_ A: Probably not - settlement has to take place on-chain anyway, so this method call could be subsumed into the first step of the settlement process.
+
+## Let's try to minimize storage
+
+Storage costs on-chain are high and tend to dwarf other gas fees. We therefore start by trying to minimize on-chain storage as much as possible.
+
+As seen in the spec, ForceMove requires certain data to be available on-chain.
+In a naive implementation we would expect something like the following to be stored for each channel:
+
+```
+struct ChannelStorage {
+  uint256 turnNumRecord;
+  uint256 finalizesAt;
+  State storedState;
+  address challengerAddress;
+}
+
+mapping(address => ChannelStatus) channelStatuses;
```

Comment:
No, I don't think it does (and by not shortening it we save some gas). Good catch! 

Thinking about this, it's probably worth having a distinction between `channelId` (used for `ForceMove`) and the `channelAddress` (used for payouts). The `channelId` should always be the hash of the `participants` and `nonce` (and `chainId`); the way the `channelAddress` depends on the settlement layer (e.g. if we were using counterfactual addressing it would be the counterfactual address and not the last 20 bytes of the `channelId`).

Reply:
LT: channelId should be bytes32